### PR TITLE
Gitlab source

### PR DESCRIPTION
**Pull request by bpatyi**
I add new gitlab source to the project.

I created the pr as you asked @dhui

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -37,17 +36,16 @@ require (
 	github.com/smartystreets/goconvey v0.0.0-20180222194500-ef6db91d284a // indirect
 	github.com/stretchr/testify v1.3.0 // indirect
 	github.com/tidwall/pretty v0.0.0-20180105212114-65a9db5fad51 // indirect
+	github.com/xanzy/go-gitlab v0.15.0
```
Comment (dhui) [inline go.mod]: Resolve conflicts and regenerate using `go mod tidy`

File: source/gitlab/gitlab.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+	}
+
+	gn := &Gitlab{
+		client:     gitlab.NewClient(nil, password),
+		url:        url,
+		migrations: source.NewMigrations(),
+	}
+
+	if u.Host != "" {
+		err = gn.client.SetBaseURL("https://" + u.Host)
```
Comment (dhui) [inline source/gitlab/gitlab.go]: Use [net/url](https://golang.org/pkg/net/url/#URL) to build the string

File: go.mod (hunk 1 above)
Comment (bpatyi) [reply go.mod]: I fixed it.

File: source/gitlab/gitlab.go (hunk 2 above)
Comment (bpatyi) [reply source/gitlab/gitlab.go]: I fixed it too.

**Pull request merged**